File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    private double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }
    
    public double getBaseEff() {
        return sideLen; // definitionally sidelen
    }

    public double getHeightEff() {
        return sideLen; // definitionally sidelen 
    }
}

File: typescript/src/shapes/square.ts
import { Shape } from "./shape.js";

function square(sideLen: number): Shape {
    return {
        getBaseEff: () => sideLen,
        getHeightEff: () => sideLen,
        getArea: function (): number {
            return this.getBaseEff() * this.getHeightEff();
        }
    };
}

export { square }